### PR TITLE
set keypad icon colour depending on state

### PR DESCRIPTION
**Pull request by sumnerboy12**
display state in key code placeholder
allow placeholder to be updated via keycode command

closes https://github.com/austinscreations/OXRS-AC-TP32-ESP-FW/issues/28

### CODE REVIEW

**Comment by sumnerboy12** [comment]
I have been testing this and it is all working, except for the rendering of the placeholder text after it gets updated (see screenshot below).

The initial `enter code` is rendered fine, but as soon as I update it (either via `keyCode` command payload, or static text) it displays as shown below.

Any ideas what is going on here?

snapshot

**Comment by sumnerboy12** [comment]
Also, I can't figure out to display the current tile "on" colour in the key code tile. Perhaps we need to add `colorRgb` to the `keyCode` command (optional) which allows you to specify the icon colour depending on locked/unlocked state (defaults to `colorOn`/`white`).

File: src/classes/classTile.cpp [hunk 1]
```diff
@@ -206,6 +206,18 @@ void classTile::setState(bool state)
     state == false ? lv_obj_clear_state(_btnDown, LV_STATE_CHECKED) : lv_obj_add_state(_btnDown, LV_STATE_CHECKED);
 }
 
+lv_color_t classTile::getColor()
```
Comment (sumnerboy12) [inline src/classes/classTile.cpp:209]: Is there a better way to get the "on" colour?

**Comment by moinmoin-sh** [comment]
> I have been testing this and it is all working, except for the rendering of the placeholder text after it gets updated (see screenshot below).
> 
> The initial `enter code` is rendered fine, but as soon as I update it (either via `keyCode` command payload, or static text) it displays as shown below.
> 
> Any ideas what is going on here?
> 
> snapshot


That has to do with my special font for the code display to show the large filled circles when hiding the pass code. It supports numericals only. You can comment out  line 57 
` lv_obj_set_style_text_font(pwd_ta, &pwd_fond_15, LV_PART_MAIN); // V
`
to get the default font which enables you to change and properly display the placeholder text. The large circles then are small dots.

**Comment by sumnerboy12** [comment]
Thanks Moin, I just tried your suggestion and it works fine now. I have posed the question on Discord to see if the others are happy to revert to these smaller dots. Do you have a strong preference?

**Comment by moinmoin-sh** [comment]
The latest commit enables the large bullets together with the dynamic placeholder text.

**Comment by sumnerboy12** [comment]
@moinmoin-sh I have rebased this after your latest changes and squashed everything down to one commit. Ready for a final review. 

As per the Discord discussions, my proposal is to leave this change as-is - i.e. the `keyPad` command allows you to send custom text to be displayed in the key code placeholder.

Once merged you can add your new fixed label which displays the parent tile text.

**Review by moinmoin-sh — APPROVED** [review]

**Pull request merged**